test/ref_ops_test.cpp

```diff
@@ -2595,13 +2664,16 @@ TEST_CASE(not_test)
         migraphx::program p;
         auto* mm = p.get_main_module();
         migraphx::shape s{migraphx::shape::int32_type, {4}};
-        auto l1 = mm->add_literal(migraphx::literal{s, {0, 8, 1, -32}});
+        std::vector<float> data{0, 8, 1, -32};
+        auto l1 = mm->add_literal(migraphx::literal{s, data});
         mm->add_instruction(migraphx::make_op("not"), l1);
         p.compile(migraphx::ref::target{});
         auto result = p.eval({}).back();
         std::vector<char> results_vector;
         result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
-        std::vector<char> gold = {1, 0, 0, 0};
+        std::vector<bool> gold(data.size());
+        std::transform(
+            data.begin(), data.end(), gold.begin(), [](const float& n) -> bool { return !n; });
```

Comment:
Dont pass by reference. Also, this should probably be `n == 0` instead of `!n` to fix the tidy warning.

Reply:
If i use `n==0` then, clang reports error that comparison of float values with `==` or `!=` is unsafe. Keeping it as `!n`.